### PR TITLE
EA-1972 Downloads List Identifier implementation

### PR DESCRIPTION
**Pull request by SrishtiSingh-eu**

### CODE REVIEW

File: src/main/java/eu/europeana/downloads/GetRecordQuery.java [hunk 1]
```diff
@@ -71,7 +65,7 @@ private void execute(OAIPMHServiceClient oaipmhServer, String currentIdentifier)
                 ZipUtility.writeInZip(zout, writer, record);
 
             }
-        } catch (IOException e) {
+        } catch (Exception e) {
```
Comment (P-Ehlert) [inline src/main/java/eu/europeana/downloads/GetRecordQuery.java:68]: It's best to not catch Exception as this also includes things like OutOfMemory Better use something like catch one or more specifc exception or if you really want to be more generic catch RuntimeException e (see also http://sonarqube.eanadev.org:9000/coding_rules?open=squid%3AS2221&rule_key=squid%3AS2221)

File: src/main/java/eu/europeana/downloads/ListIdentifierExecutor.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * Method to harvest identifiers retreived for each set. Creates a zip folder with the setname
+     * in the specified location and generates md5 sum file for the zip.
+     *
+     * @param identifiers list of identifiers
+     * @param setIdentifier setName
+     * @return number of errors occurred during the process to terminate the thread in progress.
+     */
+    private int harvestIdentifiers(List<String> identifiers, String setIdentifier) {
```
Comment (P-Ehlert) [inline src/main/java/eu/europeana/downloads/ListIdentifierExecutor.java:87]: I am a bit confused about the description; it says it harvests identifiers, but looking at the code it seems you already have the identifiers and are actually harvesting the actual records (retrieved via getRecord)?

File: src/main/java/eu/europeana/downloads/ListIdentifiersQuery.java [hunk 3]
```diff
@@ -97,37 +94,46 @@ public void execute(OAIPMHServiceClient oaipmhServer) {
 
     private void executeSingleThreadListRecord(OAIPMHServiceClient oaipmhServer, String setIdentifier) {
         List<String> identifiers = getIdentifiers(oaipmhServer, setIdentifier);
```
Comment (P-Ehlert) [inline src/main/java/eu/europeana/downloads/ListIdentifiersQuery.java:96]: For now it's fine, but to reduce complexity perhaps it's better to always do harvesting the same way (multi-threaded). If we only need to retrieve 1 set we can simply use the same approach but initialize the threadpool with 1 thread.

File: src/main/java/eu/europeana/downloads/ListRecordsExecutor.java [hunk 4]
```diff
@@ -1,73 +0,0 @@
-package eu.europeana.downloads;
-
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-
-import java.util.List;
-import java.util.concurrent.Callable;
-
-public class ListRecordsExecutor implements Callable<ListRecordsResult> {
-
```
Comment (P-Ehlert) [inline src/main/java/eu/europeana/downloads/ListRecordsExecutor.java:10]: I guess this isn't used by the ListRecords method?

Comment (SrishtiSingh-eu) [reply src/main/java/eu/europeana/downloads/ListRecordsExecutor.java:10]: Yes, this is not used by ListRecord. the naming was quite confusing. hence I created a new class ListIdentifierExecutor

File: src/main/java/eu/europeana/downloads/GetRecordQuery.java (hunk 1 above)
Comment (SrishtiSingh-eu) [reply src/main/java/eu/europeana/downloads/GetRecordQuery.java:68]: I don't think we need to catch anything there. Previously there was a ZipOutputStream and OutputStreamWriter in the try block. Now as I have removed that. We can remove this try-catch block as well

File: src/main/java/eu/europeana/downloads/ListIdentifierExecutor.java (hunk 2 above)
Comment (SrishtiSingh-eu) [reply src/main/java/eu/europeana/downloads/ListIdentifierExecutor.java:87]: Yes, I will change the description here.

File: src/main/java/eu/europeana/downloads/ListIdentifiersQuery.java (hunk 3 above)
Comment (SrishtiSingh-eu) [reply src/main/java/eu/europeana/downloads/ListIdentifiersQuery.java:96]: made the changes. Initiated thread pool with one thread and downloading a single set with the same approach

**Comment by SrishtiSingh-eu** [comment]
Resolved conflicts. Merged the code

**Pull request merged**